File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string filtered;
        for(char i: s){
            if(isalnum(i)){
                filtered+=tolower(i);
            }
        }

        string x=filtered;
        reverse(x.begin(),x.end());
        return (x==filtered? true:false);
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        x = 0
        y = len(s)-1

        while(x<y):
            while(x < y and not s[x].isalnum()):
                x+=1
            
            while(x<y and not s[y].isalnum()):
                y-=1
            
            if(s[x].lower() != s[y].lower()):
                return False
            x+=1
            y-=1
        
        return True

        